### DISCUSSION TITLE
rock scissors paper java - question

### DISCUSSION THREAD

**Discussion by Opengl555** [community/community#34647] [Other Feature Feedback, Questions, & Ideas]
public class WhoWin {
      public static String rps(String p1, String p2) {
         string rock = "rock" ;
         string scissors = "scissors";
         string paper = "paper";
       

       if(p1==rock && p2==scissors){
             System.out.println("Player 1 won!");
      }
       if(p1==scissors && p2==paper){
             System.out.println("Player 1 won!")
      }
       if(p1==paper && p2==rock){
             System.out.println("Player 1 won!")
       if(p2==rock && p1 == scissors){
             System.out.println("Player 2 won!")
      }
       if(p2==scissors && p1==paper){
             System.out.println("Player 2 won!")
      }
       if(p2==paper && p2==scissors){
             System.out.println("Player 2 won!")
      }
       if(p1==scissors && p2==scissors){
             System.out.println("Draw!")
      }
       if(p1==paper && p2 == paper){
             System.out.println("Draw!")
      }
       if(p1==rock && p2 == rock){
             System.out.println("Draw!");
      }
       return null;
     }}

   @Test
      public void test1() {
      assertEquals("Player 1 won!", WhoWin.rps("rock", 
      "scissors"));
   ......
   ......
     i wanna test it by ./build.sh ~java. im newbie ...
    can you guys give a little help on my code? i'm wanna know it so hard ....

**Comment by airtower-luna** [comment]
For being able to test your code, the important thing is that it needs to *return* the result instead of writing it standard output. Otherwise you'd have to capture the output in the test, and then it gets messy.

I'd prefer to return a simple value instead of a string, too: An enum (with three values for draw or one player winning) or if you haven't learned about enums yet an `int` (think 0 for draw, player number for the winner).

Also for comparing strings you should use [`String.equals()`](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/String.html#equals(java.lang.Object)) instead of `==`.

And finally, to check for a draw you need only one conditional, not three. Think about it for a bit. :wink:

**Reply by Opengl555** [reply]
I resolved it !

String with capital s
Systemout~ to return
==operator to equals()
and relocate @test methods into WhoWin class

Thanks so much ❤️
